File: ArrayDataMahasiswa.java
import java.util.Scanner;

public class ArrayDataMahasiswa {

    static final int MAX = 10;
    static String[] nim = new String[MAX];
    static String[] nama = new String[MAX];
    static int count = 0;

    static void show() {
        for (int i = 0; i < count; i++) {
            System.out.println(i + " " + nim[i] + " | " + nama[i]);
        }
    }

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        while (true) {
        System.out.println();
        System.out.println("==================");
        System.out.println(" MENU PROGRAM");
        System.out.println("==================");
        System.out.println("""
1 Insert Beginning
2 Insert Position
3 Insert End
4 Delete Beginning
5 Delete Position
6 Delete End
7 Delete First Occurrence
8 Show
9 Exit
""");

            System.out.print("Menu: ");
            int choice = sc.nextInt();
            sc.nextLine();

            if (choice == 1) {

                if (count == MAX) {
                    System.out.println("!ARRAY SUDAH PENUH!");
                    continue;
                }

                for (int i = count; i > 0; i--) {
                    nim[i] = nim[i-1];
                    nama[i] = nama[i-1];
                }

                System.out.print("NIM: ");
                nim[0] = sc.nextLine();
                System.out.print("Nama: ");
                nama[0] = sc.nextLine();

                count++;

            System.out.println("!!DATA BERHASIL DISIMPAN!!");

            }

            else if (choice == 2) {

            if (count == MAX) {
                    System.out.println("!ARRAY SUDAH PENUH!");
                    continue;
                }

            System.out.print("Posisi: ");
            int pos = sc.nextInt();
            sc.nextLine();

            if (pos < 0 || pos > count) {
                System.out.println("Posisi tidak valid!");
                continue;
            }

            for (int i = count; i > pos; i--) {
                nim[i] = nim[i-1];
                nama[i] = nama[i-1];
            }

            System.out.print("NIM: ");
            nim[pos] = sc.nextLine();

            System.out.print("Nama: ");
            nama[pos] = sc.nextLine();

            count++;

            System.out.println("!!DATA BERHASIL DISIMPAN!!");
        }


            else if (choice == 3) {

            if (count == MAX) {
                    System.out.println("!ARRAY SUDAH PENUH!");
                    continue;
                }

            System.out.print("NIM: ");
            nim[count] = sc.nextLine();
            System.out.print("Nama: ");
            nama[count] = sc.nextLine();

            count++;

            System.out.println("!!DATA BERHASIL DISIMPAN!!");

            }

            else if (choice == 4) {

                if (count == 0) {
                    System.out.println("!!DATA MASIH KOSONG!!");
                    continue;
                }

                for (int i = 0; i < count-1; i++) {
                    nim[i] = nim[i+1];
                    nama[i] = nama[i+1];
                }

                count--;
                System.out.println("!!DATA BERHASIL DIHAPUS!!");
            }

            else if (choice == 5) {

                if (count == 0) {
                    System.out.println("!!DATA MASIH KOSONG!!");
                    continue;
                }

                System.out.print("Posisi: ");
                int pos = sc.nextInt();

                for (int i = pos; i < count-1; i++) {
                    nim[i] = nim[i+1];
                    nama[i] = nama[i+1];
                }

                count--;
                System.out.println("!!DATA BERHASIL DIHAPUS!!");

            }

            else if (choice == 6) {

                if (count == 0) {
                    System.out.println("!!DATA MASIH KOSONG!!");
                    continue;
                }

                count--;

                System.out.println("!!DATA BERHASIL DIHAPUS!!");
            }

            else if (choice == 7) {

            if (count == 0) {
                    System.out.println("!!DATA MASIH KOSONG!!");
                    continue;
                }

            System.out.print("NIM: ");
            String target = sc.nextLine();

            int found = -1;

            for (int i = 0; i < count; i++) {
                if (nim[i].equals(target)) {
                    found = i;
                    break;
                }
            }

            if (found != -1) {

                for (int i = found; i < count-1; i++) {
                    nim[i] = nim[i+1];
                    nama[i] = nama[i+1];
                }

                count--;
                System.out.println("!!DATA BERHASIL DIHAPUS!!");

            } else {
                System.out.println("NIM tidak ditemukan!");
            }
        }

            else if (choice == 8) {
            System.out.println("== DATA MAHASISWA ==");
                show();
            }

            else if (choice == 9) {
            
            System.out.println("!!PROGRAM DIAKHIRI!!");
                break;
            }
        }
    }
}

File: ArrayDataMahasiswa.py
MAX = 10

nim = [""] * MAX
nama = [""] * MAX
count = 0


def show():

    for i in range(count):
        print(f"{i}. {nim[i]} | {nama[i]}")


while True:

    print("""
==================
 MENU PROGRAM
==================
1 Insert Beginning
2 Insert Position
3 Insert End
4 Delete Beginning
5 Delete Position
6 Delete End
7 Delete First Occurrence
8 Show
9 Exit
""")

    choice = int(input("Menu: "))

    # INSERT BEGINNING
    if choice == 1:

        if count == MAX:
            print("!ARRAY SUDAH PENUH!")
            continue

        for i in range(count, 0, -1):
            nim[i] = nim[i-1]
            nama[i] = nama[i-1]

        nim[0] = input("NIM: ")
        nama[0] = input("Nama: ")

        count += 1
        print("!!DATA BERHASIL DISIMPAN!!")

    # INSERT POSITION
    elif choice == 2:

        if count == MAX:
            print("!ARRAY SUDAH PENUH!")
            continue

        pos = int(input("Posisi: "))

        if pos < 0 or pos > count:
            print("Posisi tidak valid!")
            continue

        for i in range(count, pos, -1):
            nim[i] = nim[i-1]
            nama[i] = nama[i-1]

        nim[pos] = input("NIM: ")
        nama[pos] = input("Nama: ")

        count += 1
        print("!!DATA BERHASIL DISIMPAN!!")

    # INSERT END
    elif choice == 3:

        if count == MAX:
            print("!ARRAY SUDAH PENUH!")
            continue

        nim[count] = input("NIM: ")
        nama[count] = input("Nama: ")

        count += 1
        print("!!DATA BERHASIL DISIMPAN!!")

    # DELETE BEGINNING
    elif choice == 4:

        if count == 0:
            print("!!DATA MASIH KOSONG!!")
            continue

        for i in range(count-1):
            nim[i] = nim[i+1]
            nama[i] = nama[i+1]

        count -= 1
        print("!!DATA BERHASIL DIHAPUS!!")

    # DELETE POSITION
    elif choice == 5:

        if count == 0:
            print("!!DATA MASIH KOSONG!!")
            continue

        pos = int(input("Posisi: "))

        if pos < 0 or pos >= count:
            print("Posisi tidak valid!")
            continue

        for i in range(pos, count-1):
            nim[i] = nim[i+1]
            nama[i] = nama[i+1]

        count -= 1
        print("!!DATA BERHASIL DIHAPUS!!")

    # DELETE END
    elif choice == 6:

        if count == 0:
            print("!!DATA MASIH KOSONG!!")
            continue

        count -= 1
        print("!!DATA BERHASIL DIHAPUS!!")

    # DELETE FIRST OCCURRENCE
    elif choice == 7:

        if count == 0:
            print("!!DATA MASIH KOSONG!!")
            continue

        target = input("NIM: ")

        found = -1

        for i in range(count):
            if nim[i] == target:
                found = i
                break

        if found != -1:
            for i in range(found, count-1):
                nim[i] = nim[i+1]
                nama[i] = nama[i+1]

            count -= 1
            print("!!DATA BERHASIL DIHAPUS!!")
        else:
            print("NIM tidak ditemukan!")

    # SHOW
    elif choice == 8:
        show()

    # EXIT
    elif choice == 9:
        print("!!PROGRAM DIAKHIRI!!")
        break
